sklearn/semi_supervised/_self_training.py

```diff
@@ -613,3 +617,9 @@ def get_metadata_routing(self):
             ),
         )
         return router
+
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        if self.estimator is not None:
```

Comment:
We might want to add a `# TODO(1.8): remove together with base_estimator`.
Basically, we will remove the `None` option for `estimator` so the `if` will not be necessary.

Reply:
done in c653834